02-peer-protocol.md

```diff
@@ -1483,9 +1484,10 @@ A node:
       - SHOULD send an `error` to request the peer to fail the channel.
       - SHOULD store `my_current_per_commitment_point` to retrieve funds
         should the sending node broadcast its commitment transaction on-chain.
-    - otherwise (`your_last_per_commitment_secret` or `my_current_per_commitment_point`
-    do not match the expected values):
-      - SHOULD send an `error` and fail the channel.
+    - if `your_last_per_commitment_secret` does not match the expected values:
+      - SHOULD send an `error` and fail the channel (the sending node is lying).
+    - otherwise:
```

Comment:
I think this extra clause is out of place? Or is it meant to be a final `else` clause which was created above with "if `option_static_remotekey`"?

Reply:
It is currently under the "otherwise, if it supports `option_data_loss_protect`", which now contains three separate cases:

- your peer is honest and proves to you that you're late: send an `error` and wait for them to force-close to recover your funds
- your peer tried to tell you you were late, but they lied: force-close on them, they're fishy
- otherwise, just send an error

But this is wrong, because that otherwise also includes the case where both nodes are correctly up-to-date, so we shouldn't tell peers to send an error in that case...

More generally, I'm thinking that I should instead more heavily rework the requirements to be almost pseudo-code, based on what eclair currently does. This way it should be obvious to transform it to code in any language, and we'd be sure we have exactly the same behavior. I'll try that approach in a separate PR (it won't be trivial to review!).